### ISSUE TITLE
APIs for standup feature

### ISSUE THREAD

**Issue by chrchtjn** [Real-Dev-Squad/website-backend#977]
## Parent Issue
https://github.com/Real-Dev-Squad/website-status/issues/449
https://github.com/Real-Dev-Squad/todo-action-items/issues/134

### APIs to be made :-
1. API to mark people who have to give standup (people mandatory for standup)
2. API to save the standup provided from the status site
3. API to get all the standups given in 24 hours

**Comment by RitikJaiswal75** [comment]
# Mark User for Standup API Specification

## URL
`/users/:id/standup`

## Method
`POST`

## URL Params
- `id`: [integer] (required) - The ID of the user to be marked for standup.

## Request Body
- None

## Success Response
- Code: `200 OK`
- Content: `{ "success": true, "message": "User marked for standup successfully." }`

## Error Response
- Code: `400 Bad Request`
- Content: `{ "success": false, "message": "Invalid user ID." }`

## Notes
- This API endpoint marks the specified user for standup by updating the user model in the database.

**Comment by RitikJaiswal75** [comment]
# Store Standup Data API Specification

## URL
`/standup`

## Method
`POST`

## URL Params
- None

## Request Body
- `timestamp`: [string] (required) - The current date and time in ISO 8601 format.
- `yesterday`: [string] (required) - A description of what the user did yesterday.
- `today`: [string] (required) - A description of what the user plans to do today.
- `blockers`: [string] (optional) - A description of any blockers the user is facing.
- `user_id`: [integer] (required) - The ID of the user submitting the standup.

## Success Response
- Code: `200 OK`
- Content: `{ "success": true, "message": "Standup data stored successfully." }`

## Error Response
- Code: `400 Bad Request`
- Content: `{ "success": false, "message": "Invalid request body." }`
- Code: `401 Unauthorized`
- Content: `{ "success": false, "message": "User not authenticated." }`

## Notes
- This API endpoint stores standup data submitted by the user in the database.

**Comment by RitikJaiswal75** [comment]
# Retrieve User Standup Data API Specification

## URL
`/standup/:user_id`

## Method
`GET`

## URL Params
- `user_id`: [integer] (required) - The ID of the user whose standup data is being retrieved.

## Request Body
- None

## Success Response
- Code: `200 OK`
- Content: `{ "success": true, "data": { "timestamp": "2023-04-20T16:23:58Z", "yesterday": "Worked on feature X", "today": "Will work on feature Y", "blockers": "None", "user_id": 123 } }`

## Error Response
- Code: `400 Bad Request`
- Content: `{ "success": false, "message": "Invalid user ID." }`
- Code: `404 Not Found`
- Content: `{ "success": false, "message": "Standup data not found for the specified user." }`

## Notes
- This API endpoint retrieves standup data for the specified user from the database.

**Comment by heyrandhir** [comment]
For `/standup/:id` we can also add one more path param which signifies the marking or unmarking.  e.g `/standup/<userId:string>/<isMarked:boolean>`.

**Comment by heyrandhir** [comment]
> ### Navigation
> [Overview](https://github.com/Real-Dev-Squad/website-backend/issues/977#issue-1647906543)	[Mark API](https://github.com/Real-Dev-Squad/website-backend/issues/977#issuecomment-1516612562)	[Save Standup API](https://github.com/Real-Dev-Squad/website-backend/issues/977#issuecomment-1516619881)	Get Standup API
> # Retrieve User Standup Data API Specification
> ## URL
> `/standup/:id`
> 
> ## Method
> `GET`
> 
> ## URL Params
> * `id`: [string] (required) - The ID of the user whose standup data is being retrieved.
> 
> ## Request Body
> * None
> 
> ## Success Response
> * Code: `200 OK`
> * Content: `{ "success": true, "data": { "timestamp": "2023-04-20T16:23:58Z", "yesterday": "Worked on feature X", "today": "Will work on feature Y", "blockers": "None", "user_id": 123 } }`
> 
> ## Error Response
> * Code: `400 Bad Request`
> * Content: `{ "success": false, "message": "Invalid user ID." }`
> * Code: `404 Not Found`
> * Content: `{ "success": false, "message": "Standup data not found for the specified user." }`
> 
> ## Notes
> * This API endpoint retrieves standup data for the specified user from the database.
> 
> ### Navigation
> [Overview](https://github.com/Real-Dev-Squad/website-backend/issues/977#issue-1647906543)	[Mark API](https://github.com/Real-Dev-Squad/website-backend/issues/977#issuecomment-1516612562)	[Save Standup API](https://github.com/Real-Dev-Squad/website-backend/issues/977#issuecomment-1516619881)	Get Standup API

We can add a path param to specify the dates between which we want the updates. this will reduce the load on backed.

**Comment by RitikJaiswal75** [comment]
> We can add a path param to specify the dates between which we want the updates. this will reduce the load on backed. additionally this route should only be authorized to superuser or author of the standup.

I don't think that would need to go in route. Also if i give a reply that consists all the standups that would make the API slow. I plan on creating a feature in future that would return the dates and once the date will be clicked it can resolve to the standup of specific date

>For /standup/:id we can also add one more path param which signifies the marking or unmarking. e.g /standup/<userId:string>/<isMarked:boolean>

A post request on /standup/:id would do that. that is what the mark api does

**Comment by DashDeipayan** [comment]
Looks good to me

**Comment by bhtibrewal** [comment]
looks okay to me
